pygmt/src/shift_origin.py

```diff
@@ -2,20 +2,42 @@
 shift_origin - Shift plot origin in x and/or y directions.
 """
 
+from contextlib import contextmanager
+
 from pygmt.clib import Session
+from pygmt.helpers import build_arg_list
 
 
 def shift_origin(
     self, xshift: float | str | None = None, yshift: float | str | None = None
 ):
     r"""
-    Shift plot origin in x and/or y directions.
+    Shift the plot origin in x and/or y directions.
+
+    The shifts can be temporary or permanent. If used as a context manager, the shifts
+    are temporary and only apply to the block of code within the context manager. If
+    used as a standalone method, the shifts are permanent and apply to all subsequent
+    plots.
+
+    1.  Use as a context manager to shift the plot origin temporarily:
+
+        .. code-block:: python
+
+            with fig.shift_origin(...):
+                ...  # Other plot commands
+                ...
+
+    2.  Use as a standalone method to shift the plot origin permanently:
+
+        .. code-block:: python
+
+            fig.shift_origin(xshift=12)
+            ...  # Other plot commands
 
-    This method shifts the plot origin relative to the current origin by *xshift* and
-    *yshift* in x and y directions, respectively. Optionally, append the length unit
+    The shifts *xshift* and *yshift* in x and y directions are relative to the current
+    plot origin. The default unit for shifts is centimeters (**c**) but can be changed
+    to other units via :gmt-term:`PROJ_LENGTH_UNIT`. Optionally, append the length unit
     (**c** for centimeters, **i** for inches, or **p** for points) to the shifts.
-    Default unit if not explicitly given is **c**, but can be changed to other units via
-    :gmt-term:`PROJ_LENGTH_UNIT`.
 
     For *xshift*, a special character **w** can also be used, which represents the
     bounding box **width** of the previous plot. The full syntax is
```

Comment:
How do **w** (x direction) and **h** (y direction) work for the temporary option? Is this also refering to the previous plot, or actually the current plot? In case it refers to the previous plot, what is used if the current plot is the first plot?

Reply:
> How do **w** (x direction) and **h** (y direction) work for the temporary option? Is this also refering to the previous plot, or actually the current plot? 

It always refers to the previous plot.

> In case it refers to the previous plot, what is used if the current plot is the first plot?

`Figure.shift_origin` has no effects if it is called before any plotting methods:
```
import pygmt

fig = pygmt.Figure()
with fig.shift_origin(xshift="w+2c"):
    fig.basemap(region=[0, 1, 0, 1], projection="X3c", frame=0)
fig.show()
```